src/core_modules/capture-core/rules/postProcessRulesEffects.js

```diff
@@ -11,47 +11,41 @@ const getAssignEffectsBasedOnHideField = (hideEffects: Array<HideOutputEffect>)
             type: effectActions.ASSIGN_VALUE,
         }));
 
-const getAssignEffectsBasedOnHideSection = (
-    hideSectionEffects: Array<HideOutputEffect>,
-    foundation: RenderFoundation,
-) => hideSectionEffects
-    .flatMap(({ id: sectionId }) => {
-        const section = foundation.getSection(sectionId);
-        if (!section) {
-            return [];
-        }
-
-        return [...section.elements.values()]
-            .map(({ id }) => ({
-                id,
-                value: null,
-                type: effectActions.ASSIGN_VALUE,
-            }));
-    });
+const deduplicateEffectArray = (effectArray: Array<OutputEffect>) => {
+    const dedupedEffectsAsMap = new Map(effectArray.map(effect => [effect.id, effect]));
+    return [...dedupedEffectsAsMap.values()];
+};
 
 const postProcessAssignEffects = ({
     assignValueEffects,
     hideFieldEffects,
-    hideSectionEffects,
-    foundation,
 }: {
     assignValueEffects: Array<AssignOutputEffect>,
     hideFieldEffects: Array<HideOutputEffect>,
-    hideSectionEffects: Array<HideOutputEffect>,
-    foundation: RenderFoundation,
-}) => {
-    const fromHideFieldArray = getAssignEffectsBasedOnHideField(hideFieldEffects);
-    const fromHideSectionArray = getAssignEffectsBasedOnHideSection(hideSectionEffects, foundation);
-
-    const dedupedEffectsAsMap = new Map([
+}) => (
+    // assignValueEffects has precedence over "blank a hidden field"-assignments.
+    // This requirement is met by destructuring assignValueEffects first.
+    deduplicateEffectArray([
         ...assignValueEffects,
-        ...fromHideFieldArray,
-        ...fromHideSectionArray,
-    ].map(effect => [effect.id, effect]));
+        ...getAssignEffectsBasedOnHideField(hideFieldEffects),
+    ])
+);
 
+const postProcessHideSectionEffects = (
+    hideSectionEffects: Array<HideOutputEffect>,
+    foundation: RenderFoundation,
+) => (hideSectionEffects.flatMap(({ id: sectionId }) => {
```

Comment:
What happens here is that the map-part of `flatMap` maps each section to an array of hide field effects, and then the flat-part spreads each of these subarrays. I had to read the code again to see this. It would probably be more transparent if I just use `reduce` instead.

Reply:
Ah, I misread this. Let's keep this as is, I like it 👍

Reply:
Oops, too late...

Reply:
Reverted it.